modules/@ergonode/dashboard/src/pages/dashboard/index.vue

```diff
@@ -18,44 +18,45 @@ import EmptyDashboardPage from '@Dashboard/components/Pages/EmptyDashboardPage';
 import {
     getProductsCount,
 } from '@Dashboard/services';
+import PRODUCT_PRIVILEGES from '@Products/config/privileges';
 
 export default {
     name: 'Dashboard',
     components: {
         DashboardPage,
         EmptyDashboardPage,
     },
-    async asyncData({
-        app,
-    }) {
-        let productsCount = 0;
-
-        try {
-            productsCount = await getProductsCount({
-                $axios: app.$axios,
-            });
-        } catch (e) {
-            if (app.$axios.isCancel(e)) {
-                return {
-                    productsCount,
-                };
-            }
-        }
-
+    async fetch() {
+        await this.onProductCreated();
```

Comment:
What's the reason about changing hooks? The same thing might be done with asyncData. Right now user will enter dashboard and might see layout jumps without any pre loader

Reply:
Moving the logic to a single asynchronous method, causes an error to pop up after a page refresh for `asyncData`. Additionally, in most applications, we use `fetch` so it's also a matter of consistent structure.